### PR TITLE
feat(receiver-mock): add trace storage

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
This PR adds storing traces and querying spans by their attributes to the receiver mock.

### CODE REVIEW

**Comment by aboguszewski-sumo** [comment]
I added the possibility to query the spans by their attributes. More features like querying by the name or ids, or querying traces, will be added in another pull requests.

**Review by swiatekm-sumo — COMMENTED** [review]
The changes look good to me. Can you also document the new endpoint in the README?

**Comment by aboguszewski-sumo** [comment]
I added an entry in the docs.

File: src/rust/receiver-mock/README.md [hunk 1]
```diff
@@ -36,6 +38,65 @@ Example output:
...
 
+## Traces
+
+The following endpoints provide information about received traces:
+
+- `/spans-list` - returns list of collected spans
+
+  It accepts a list of key value pairs being an attribute set that the span will
+  have to fullfil in order to be returned.
+  Attribute values can be ommitted in which case only presence of a particular attribute
```
Comment (sumo-drosiek) [inline src/rust/receiver-mock/README.md:49]:
```suggestion
  Attribute values can be omitted in which case only presence of a particular attribute
```

**Review by sumo-drosiek — APPROVED** [review]
LGTM, one comment for documentation

**Pull request merged**